### PR TITLE
Add .NET client for Dapr Jobs API

### PR DESCRIPTION
**Pull request by WhitWaldo**
# Description

Added a Dapr Jobs client for the new [Jobs API](https://v1-14.docs.dapr.io/reference/api/jobs_api/#schedule-a-job).

This PR is intended to replace the previous two (#1320 and #1331) as they both had a missing DCO and were full of obsolete references that have now been merged into the project.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1321 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [X] Extended the documentation
* [X] Added sample project

### CODE REVIEW

File: src/Dapr.Jobs/DaprJobsGrpcClient.cs [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    internal DaprJobsGrpcClient(
+        Autogenerated.Dapr.DaprClient innerClient,
+        HttpClient httpClient,
+        KeyValuePair<string, string>? apiTokenHeader)
+    {
+        this.client = innerClient;
+        this.httpClient = httpClient;
+        this.apiTokenHeader = apiTokenHeader;
+
+        this.httpClient.DefaultRequestHeaders.UserAgent.Add(UserAgent());
```
Comment (philliphoff) [inline src/Dapr.Jobs/DaprJobsGrpcClient.cs:55]: Nit: can you pass `this.userAgent`?

File: src/Dapr.Jobs/DaprJobsGrpcClient.cs [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        }
+    }
+
+    private CallOptions CreateCallOptions(Metadata? headers, CancellationToken cancellationToken)
+    {
+        var callOptions = new CallOptions(headers: headers ?? new Metadata(), cancellationToken: cancellationToken);
+
+        callOptions.Headers!.Add("User-Agent", this.userAgent);
+
+        if (apiTokenHeader is not null)
```
Comment (philliphoff) [inline src/Dapr.Jobs/DaprJobsGrpcClient.cs:216]: Nit: add braces for clarity.

File: src/Dapr.Jobs/Extensions/EndpointRouteBuilderExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            }
+
+            var parameters = new List<object?> { jobName, jobPayload };
+            var actionParameters = action.Method.GetParameters().Skip(2).ToArray();
+            parameters.AddRange(actionParameters
+                .Where(parameter => parameter.ParameterType != typeof(CancellationToken))
+                .Select(parameter => context.RequestServices.GetService(parameter.ParameterType)));
+            parameters.Add(cancellationToken);
+            
+            var result = action.DynamicInvoke(parameters.ToArray());
```
Comment (philliphoff) [inline src/Dapr.Jobs/Extensions/EndpointRouteBuilderExtensions.cs:69]: Snazzy, but could result in strange runtime errors if/when the user does not abide by the assumptions of the parameter order.  Would it be not much more difficult to allow *any* order (as is generally allowed by ASP.NET)?

E.g.:

```csharp
var parameters =
    action
        .Method
        .GetParameters()
        .Select(
            parameter => parameter.ParameterType switch
            {
                System.String => jobName,
                DaprJobDetails => daprJobDetails,
                CancellationToken => cancellationToken,
                _ => context.RequestServices.GetService(parameter.ParameterType)
            }
        .ToArray();
```

File: src/Dapr.Jobs/Models/DaprJobSchedule.cs [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+/// <summary>
+/// Used to build a schedule for a job.
+/// </summary>
+[JsonConverter(typeof(DaprJobScheduleConverter))]
+public sealed class DaprJobSchedule
+{
+    /// <summary>
+    /// A regular expression used to evaluate whether a given prefix period embodies an @every statement.
+    /// </summary>
+    private readonly Regex isEveryExpression = new(@"^@every (\d+(m?s|m|h))+$", RegexOptions.Compiled);
```
Comment (philliphoff) [inline src/Dapr.Jobs/Models/DaprJobSchedule.cs:30]: `static`?  (Here and next property below.)

**Review by philliphoff — APPROVED** [review]
Some suggestions and other final nits, but otherwise looks good.

File: src/Dapr.Jobs/DaprJobsGrpcClient.cs (hunk 1 above)
Comment (WhitWaldo) [reply src/Dapr.Jobs/DaprJobsGrpcClient.cs:55]: Looking at it, we don't even need that line. It's covered by the later invocation of `CreateCallOptions` to set up the various `CallOptions` on each request which in turn is using `this.userAgent`.

File: src/Dapr.Jobs/DaprJobsGrpcClient.cs (hunk 2 above)
Comment (WhitWaldo) [reply src/Dapr.Jobs/DaprJobsGrpcClient.cs:226]: Added braces throughout project

File: src/Dapr.Jobs/Models/DaprJobSchedule.cs (hunk 4 above)
Comment (WhitWaldo) [reply src/Dapr.Jobs/Models/DaprJobSchedule.cs:30]: Done

File: src/Dapr.Jobs/Extensions/EndpointRouteBuilderExtensions.cs (hunk 3 above)
Comment (WhitWaldo) [reply src/Dapr.Jobs/Extensions/EndpointRouteBuilderExtensions.cs:86]: Good point - updated to accommodate any order, including if the cancellation token is not provided by the user in their delegate (added as CancellationToken.None)

**Review by philliphoff — APPROVED** [review]

**Pull request merged**